dlt/extract/validation.py

```diff
@@ -0,0 +1,35 @@
+from typing import Optional, Protocol, TypeVar, Generic, Type, Union, Any
+
+try:
+    from pydantic import BaseModel as PydanticBaseModel, ValidationError as PydanticValidationError
+except ModuleNotFoundError:
+    PydanticBaseModel = None  # type: ignore[misc]
+
+from dlt.extract.exceptions import ValidationError
+from dlt.common.typing import TDataItem
+from dlt.common.schema.typing import TAnySchemaColumns, ColumnValidator
+from dlt.extract.typing import TTableHintTemplate
+
+
+_TPydanticModel = TypeVar("_TPydanticModel", bound=PydanticBaseModel)
+
+
+class PydanticValidator(ColumnValidator, Generic[_TPydanticModel]):
+    model: Type[_TPydanticModel]
+    def __init__(self, model: Type[_TPydanticModel]) -> None:
+        self.model = model
+
+    def __call__(self, item: TDataItem, meta: Any = None) -> _TPydanticModel:
+        """Validate a data item agains the pydantic model"""
+        if item is None:
```

Comment:
note that `item` may be a list. if that is a list, validate elements of a list. (we could also synthesize a model that validates lists of objects automatically - could we way faster)

please also test it

Reply:
Yep, it's easy to synthesize wrapper model. I assume it would be faster.   
Only tricky part is it would fail the whole list on one error, if you want to implement a "ignore invalid items" option. But think there's some ways to manage it.

Reply:
we are adding such settings in #594 so we'll upgrade pydantic validation as well

Reply:
OK - looks like will be synthesizing models depending on the settings above. so let's skip it for now. it will come back :)

Reply:
So I just added synthesized model anyway: https://github.com/dlt-hub/dlt/pull/638/files#diff-ba6e7ddb3759fcdb3e599cf5ca145432d28d529845672325b5642e4d00aa4d3dR23-R26  
You could make this conditional depending how you  want to structure it. There are also some solutions to achieve list validation without raising in https://github.com/pydantic/pydantic/issues/2274 but could need separate implementations for pydantic 1 and 2